sklearn/decomposition/_nmf.py

```diff
@@ -1330,6 +1340,12 @@ def fit(self, X, y=None, **params):
         -------
         self
         """
+        if self.init == 'warn':
```

Comment:
I think the warning should be in `non_negative_factorization` because it's public.
It might even be delegated to `_initialize_nmf`